pallets/manta-pay/src/errors.rs

```diff
@@ -0,0 +1,349 @@
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Error Handling for MantaPay
+
+use super::*;
+
+impl<T> From<InvalidAuthorizationSignature> for Error<T>
+where
+    T: Config,
+{
+    #[inline]
+    fn from(_: InvalidAuthorizationSignature) -> Self {
+        Self::InvalidAuthorizationSignature
+    }
+}
+
+impl<T> From<InvalidSourceAccount<config::Config, T::AccountId>> for Error<T>
+where
+    T: Config,
+{
+    #[inline]
+    fn from(_: InvalidSourceAccount<config::Config, T::AccountId>) -> Self {
+        Self::InvalidSourceAccount
+    }
+}
+
+impl<T> From<InvalidSinkAccount<config::Config, T::AccountId>> for Error<T>
+where
+    T: Config,
+{
+    #[inline]
+    fn from(_: InvalidSinkAccount<config::Config, T::AccountId>) -> Self {
+        Self::InvalidSinkAccount
+    }
+}
+
+/// Merkle Tree Parameters Decode Error Type
+pub type MTParametersError = codec::DecodeError<
+    <&'static [u8] as codec::Read>::Error,
+    <config::UtxoAccumulatorModel as codec::Decode>::Error,
+>;
+
+/// Utxo Accumulator Item Hash Decode Error Type
+pub type UtxoItemHashError = codec::DecodeError<
+    <&'static [u8] as codec::Read>::Error,
+    <config::utxo::UtxoAccumulatorItemHash as codec::Decode>::Error,
+>;
+
+/// Verification Context Decode Error Type
+pub type VerifyingContextError = codec::DecodeError<
+    <&'static [u8] as codec::Read>::Error,
+    <config::VerifyingContext as codec::Decode>::Error,
+>;
+
+/// Receiver Ledger Error
+pub enum ReceiverLedgerError<T>
+where
+    T: Config,
+{
+    /// Utxo Decoding Error
+    UtxoDecodeError(scale_codec::Error),
+
+    /// Wrong Checksum Error
+    ChecksumError,
+
+    /// Merkle Tree Parameters Decoding Error
+    MTParametersDecodeError(MTParametersError),
+
+    /// Utxo Accumulator Item Hash Decoding Error
+    UtxoAccumulatorItemHashDecodeError(UtxoItemHashError),
+
+    /// Merkle Tree Out of Capacity Error
+    MerkleTreeCapacityError,
+
+    /// Field Element Encoding Error
+    FpEncodeError(scale_codec::Error),
+
+    /// Field Element Encoding Error
+    FpDecodeError(scale_codec::Error),
+
+    /// Path Decoding Error
+    PathDecodeError(scale_codec::Error),
+
+    /// Full Incoming Note Decoding Error
+    FullNoteDecodeError(scale_codec::Error),
+
+    /// Asset Registered Error
+    ///
+    /// The asset has already been registered with the ledger.
+    AssetRegistered,
+
+    /// Type Marker Parameter
+    Marker(PhantomData<T>),
+}
+
+impl<T> From<ReceiverPostError<ReceiverLedgerError<T>>> for Error<T>
+where
+    T: Config,
+{
+    #[inline]
+    fn from(err: ReceiverPostError<ReceiverLedgerError<T>>) -> Self {
+        match err {
+            ReceiverPostError::AssetRegistered => Self::AssetRegistered,
+            ReceiverPostError::UnexpectedError(e) => match e {
+                ReceiverLedgerError::<T>::AssetRegistered => Self::AssetRegistered,
```

Comment:
Exhaustive error handling creates duplicates... It functionally is fine, but could change `manta-rs` to get rid of this